jenkins.go

```diff
@@ -12,95 +12,62 @@
 // License for the specific language governing permissions and limitations
 // under the License.
 
-// Gojenkins is a Jenkins Client in Go, that exposes the jenkins REST api in a more developer friendly way.
+// Package gojenkins is a Jenkins Client in Go, that exposes the jenkins REST api in a more developer friendly way.
 package gojenkins
 
 import (
-	"crypto/tls"
-	"crypto/x509"
 	"errors"
 	"fmt"
 	"log"
 	"net/http"
-	"net/http/cookiejar"
-	"net/url"
 	"os"
 	"strconv"
 	"strings"
 )
 
-// Basic Authentication
+// BasicAuth represents Basic Authentication
 type BasicAuth struct {
 	Username string
 	Password string
 }
 
+// Jenkins represents a Jenkins client connection
 type Jenkins struct {
 	Server    string
 	Version   string
 	Raw       *ExecutorResponse
 	Requester *Requester
 }
 
-// Loggers
 var (
 	Info    *log.Logger
 	Warning *log.Logger
 	Error   *log.Logger
 )
 
 // Init Method. Should be called after creating a Jenkins Instance.
-// e.g jenkins := CreateJenkins("url").Init()
+// e.g jenkins := New("url").Init()
 // HTTP Client is set here, Connection to jenkins is tested here.
 func (j *Jenkins) Init() (*Jenkins, error) {
 	j.initLoggers()
-	// Skip SSL Verification?
-	tlsCfg := &tls.Config{
-		InsecureSkipVerify: !j.Requester.SslVerify,
-	}
-	if j.Requester.CACert != nil {
-		pool := x509.NewCertPool()
-		pool.AppendCertsFromPEM(j.Requester.CACert)
-		tlsCfg.RootCAs = pool
-		// always verify certs if custom ca cert is used.
-		tlsCfg.InsecureSkipVerify = false
-	}
-	tr := &http.Transport{
-		TLSClientConfig: tlsCfg,
-	}
-
-	if j.Requester.Client == nil {
-		cookies, _ := cookiejar.New(nil)
-
-		if os.Getenv("HTTP_PROXY") != "" {
-			proxyUrl, _ := url.Parse(os.Getenv("HTTP_PROXY"))
-			tr.Proxy = http.ProxyURL(proxyUrl)
-		}
-
-		client := &http.Client{
-			Transport: tr,
-			Jar:       cookies,
-			// Function to add auth on redirect.
-			CheckRedirect: j.Requester.redirectPolicyFunc,
-		}
-
-		j.Requester.Client = client
-	}
-
+	j.Requester.Client = http.DefaultClient
```

Comment:
Yeah, that was part of a fix.  It was going to be a multi-phased approach:

1. Add `InitWithClient()`
2. Replace all occurrences of `Init` with `InitWithClient`
3. Remove `Init`
4. Rename all `InitWithClient` to `Init.

etc. etc.  More along the lines of what you get with the [Refactoring Book ](https://martinfowler.com/books/refactoring.html) techniques.